src/test/java/org/jenkinsci/plugins/apachehttpclient4api/Client4JDKInstallerTest.java

```diff
@@ -0,0 +1,156 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.apachehttpclient4api;
+
+import static org.junit.Assert.assertEquals;
+import hudson.tools.JDKInstaller.DescriptorImpl;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import hudson.model.JDK;
+import hudson.util.StreamTaskListener;
+import hudson.tools.JDKInstaller.Platform;
+import hudson.tools.JDKInstaller.CPU;
+import hudson.tools.InstallSourceProperty;
+import hudson.tools.JDKInstaller;
+import hudson.tools.ToolInstaller;
+import hudson.util.FormValidation;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.apachehttpclient4api.Client4JDKInstaller.Client4DescriptorImpl;
+import org.junit.Assert;
+import org.junit.Assume;
+
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.xml.sax.SAXException;
+
+/**
+ * Tests for {@link Client4JDKInstaller} using Apache HttpClient 4.x.
+ * @author Kohsuke Kawaguchi
+ * @author Oleg Nenashev
+ */
+public class Client4JDKInstallerTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+    
+    private boolean usingMockedCredentials;
+    
+
+    @Before
+    public void setUp() throws Exception {
+        File f = new File(new File(System.getProperty("user.home")),".jenkins-ci.org");
+        String u = null;
+        String p = null;
+              
+        if (!f.exists()) {
+            LOGGER.log(Level.WARNING, "{0} doesn''t exist. Cannot load real credentials", f);
+        } else {
+            Properties prop = new Properties();
+            try (InputStream in = Files.newInputStream(f.toPath())) {
+                prop.load(in);
+                u = prop.getProperty("oracle.userName");
+                p = prop.getProperty("oracle.password");
+                if (u==null || p==null) {
```

Comment:
Different log message. This one explicitly says it's a fallback. That `~/.jenkins-ci.org` file is being used for multiple purposes, unfortunately